pandas/tseries/tests/test_tslib.py

```diff
@@ -519,7 +519,11 @@ def test_parsers(self):
                  '2014-06': datetime.datetime(2014, 6, 1),
                  '06-2014': datetime.datetime(2014, 6, 1),
                  '2014-6': datetime.datetime(2014, 6, 1),
-                 '6-2014': datetime.datetime(2014, 6, 1), }
+                 '6-2014': datetime.datetime(2014, 6, 1),
+
+                 '20010101 12': datetime.datetime(2001, 1, 1, 12),
+                 '20010101 1234': datetime.datetime(2001, 1, 1, 12, 34),
```

Comment:
So the current version accepts `12` in addition to `1`. I was thinking we should keep the two digit form.

```
In [23]: pd.Timestamp('20010101 1')
Out[23]: Timestamp('2001-01-01 01:00:00')

In [24]: pd.Timestamp('20010101 09')
Out[24]: Timestamp('2001-01-01 09:00:00')

In [25]: pd.Timestamp('20010101 12')
Out[25]: Timestamp('2001-01-01 12:00:00')
```

As for four digits, I was thinking it would make sense to allow "military" times like `0800` (which is also currently accepted).

```
In [26]: pd.Timestamp('20010101 0800')
Out[26]: Timestamp('2001-01-01 08:00:00')
```